code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -173,5 +176,14 @@
 				step(AM,get_dir(AM,src))
 		if(get_dist(src,B.target) == 0)
 			B.End()
-		if(QDELETED(B))
-			vines -= B
+
+/**
+  * Removes a vine from the list.
+  *
+  * Removes the vine from our list.
+  * Called specifically when the vine is about to be destroyed, so we don't have any null references.
+  * Arguments:
+  * * datum/beam/vine - The vine to be removed from the list.
+  */
+mob/living/simple_animal/hostile/venus_human_trap/proc/remove_vine(datum/beam/vine)
```

Comment:
```suggestion
mob/living/simple_animal/hostile/venus_human_trap/proc/remove_vine(datum/beam/vine, force)
```
The signal sends two args: `SEND_SIGNAL(D, COMSIG_PARENT_QDELETING, force)`
The first one is always the source, so you have it right, the second one here is `force`. Just for reference.